src/npm/NpmPackage.ts

```diff
@@ -0,0 +1,91 @@
+import * as request from "request-promise-native";
+import { S3Cache } from "./S3Cache";
+import { version } from "punycode";
+
+export class NpmPackage {
```

Comment:
As much as possible, it'd be nice to have jsdoc comments for each module/class/function, for our own future reference and so we're making the best possible typedoc (`npm run docs`) pages.

Reply:
I'll change this. I didn't know it was the convention for this project.  
As a side note I'd like to mention that in most cases my personal preference is to avoid unnecessary documentation when module/class/function have descriptive names and the functionality is focused and matches the name of the containing block of code.

Reply:
Totally reasonable, feel free to use your best judgement, but assume you work with simpletons (me) that will have a hard time inferring functionality from anything non-obvious

Reply:
fixed in https://github.com/troyready/serverless-artifact-cacher/pull/3/commits/bd79d9d2e98691c0219e903884666ad37f570966